cms/static/js/views/paged_container.js

```diff
@@ -157,6 +161,19 @@ define(["jquery", "underscore", "js/views/container", "js/utils/module", "gettex
 
             sortDisplayName: function() {
                 return gettext("Date added");  // TODO add support for sorting
+            },
+
+            toggle_previews: function(){
+                var self = this,
+                    xblockUrl = this.model.url();
+                return $.ajax({
+                    // No runtime, so can't get this via the handler() call.
+                    url: '/preview' + decodeURIComponent(xblockUrl) + "/handler/trigger_previews",
```

Comment:
This is a shame. Can't see an obvious fix though, if we're keeping the state on the XBlock. I've run into similar problems before where it's really hard to make one part of the client-side code talk to another part since all the state variables are scoped very tightly.